molgenis-data-import/src/main/java/org/molgenis/data/importer/ImportApi.java

```diff
@@ -0,0 +1,214 @@
+import org.molgenis.data.support.GenericImporterExtensions;
+import org.molgenis.data.system.ImportRun;
+import org.molgenis.file.FileStore;
+import org.molgenis.security.core.utils.SecurityUtils;
+import org.molgenis.util.FileExtensionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.molgenis.data.importer.ImportApi.URI;
+
+@Controller
+@RequestMapping(URI)
+public class ImportApi
+{
+	private static final long serialVersionUID = 1L;
+	private static final Logger LOG = LoggerFactory.getLogger(ImportApi.class);
+
+	public static final String ID = "importer";
+	public static final String URI = "api/" + ID;
+
+	private ImportServiceFactory importServiceFactory;
+	private FileStore fileStore;
+	private FileRepositoryCollectionFactory fileRepositoryCollectionFactory;
+	private ImportRunService importRunService;
+	private ExecutorService asyncImportJobs;
+	private DataService dataService;
+
+	@Autowired
+	public ImportApi(ImportServiceFactory importServiceFactory, FileStore fileStore,
+			FileRepositoryCollectionFactory fileRepositoryCollectionFactory, ImportRunService importRunService,
+			DataService dataService)
+	{
+		this.importServiceFactory = importServiceFactory;
+		this.fileStore = fileStore;
+		this.fileRepositoryCollectionFactory = fileRepositoryCollectionFactory;
+		this.importRunService = importRunService;
+		this.dataService = dataService;
+		this.asyncImportJobs = Executors.newCachedThreadPool();
+	}
+
+	public ImportApi(ImportServiceFactory importServiceFactory, FileStore fileStore,
+			FileRepositoryCollectionFactory fileRepositoryCollectionFactory, ImportRunService importRunService,
+			DataService dataService, ExecutorService executorService)
+	{
+		this.importServiceFactory = importServiceFactory;
+		this.fileStore = fileStore;
+		this.fileRepositoryCollectionFactory = fileRepositoryCollectionFactory;
+		this.importRunService = importRunService;
+		this.dataService = dataService;
+		this.asyncImportJobs = executorService;
+	}
+
+	@RequestMapping(method = RequestMethod.POST, value = "/importByUrl")
```

Comment:
Okey, it will be fixed by my next commit, but a little warning up front: the url does not work in java because it does not automagically go to the https url, the http version is not available.